src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -2359,6 +2359,8 @@
          is null before type inference, and it is replaced by a BoundDiscardExpression with a non-null
          type after inference. -->
     <Field Name="Type" Type="TypeSymbol?" Override="true"/>
+    <Field Name="NullableAnnotation" Type="NullableAnnotation"/>
+    <Field Name="IsTypeDefinedExplicitly" Type="bool"/>
```

Comment:
nit: I looked for a name that is already used. `IsInferred` is used for array creations, `InferredType` is used for local declarations. Consider using one of those (personally, I like `IsInferred` since that's clearly a bool).